server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingSupplierPlugin.java

```diff
@@ -8,15 +8,13 @@
 
 package org.elasticsearch.plugins.internal;
 
-import java.util.function.Supplier;
-
 /**
- * An internal plugin that will return a supplier of DocumentParsingObserver.
+ * An internal plugin that will return a supplier of DocumentParsingSupplier.
  */
-public interface DocumentParsingObserverPlugin {
+public interface DocumentParsingSupplierPlugin {
 
     /**
-     * @return a supplier of DocumentParsingObserver to allow observing parsing events
+     * @return a DocumentParsingSupplier to create instances of observer and reporter of parsing events
      */
-    Supplier<DocumentParsingObserver> getDocumentParsingObserverSupplier();
+    DocumentParsingSupplier getDocumentParsingSupplier();
```

Comment:
Can we eliminate a level of indirection here by having the plugin interface contain the 3 methods for constructing these classes?

Reply:
the `DocumentParsingSupplier`  is being used in transport classes/ingestService. It would be odd to pass down a plugin into those classes